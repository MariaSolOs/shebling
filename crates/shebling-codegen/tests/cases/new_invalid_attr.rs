use shebling_codegen::New;

// Invalid uses of the #[new] attribute.

#[derive(New)]
struct Foo {
    #[new]
    foo: String,
}

#[derive(New)]
struct Bar {
    #[new()]
    bar: String,
}

#[derive(New)]
struct Baz {
    #[new(wrong)]
    baz: String,
}

fn main() {}
