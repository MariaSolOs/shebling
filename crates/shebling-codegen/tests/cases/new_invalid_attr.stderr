error: expected attribute arguments in parentheses: #[new(...)]
 --> tests/cases/new_invalid_attr.rs:7:7
  |
7 |     #[new]
  |       ^^^

error: unexpected end of input, expected nested attribute
  --> tests/cases/new_invalid_attr.rs:13:11
   |
13 |     #[new()]
   |           ^

error: invalid #[new(...)] content
  --> tests/cases/new_invalid_attr.rs:19:11
   |
19 |     #[new(wrong)]
   |           ^^^^^
